### ISSUE TITLE
ConfigurationPageResolver::getClosestPageIdWithActiveTemplate must be dropped

### ISSUE THREAD

**Issue by georgringer** [TYPO3-Solr/ext-solr#1475]
Even though I don't know currently the reason for that code it must be dropped. Since 8.6 there is a nice way to include typoscript without any sys_template record. See https://docs.typo3.org/typo3cms/extensions/core/Changelog/8.6/Feature-79140-AddHookToAddCustomTypoScriptTemplates.html for details.

I am already starting using this hook and others will do as well which then will lead to false positives.

Solution could be to use `\TYPO3\CMS\Core\TypoScript\TemplateService::runThroughTemplates`

**Comment by dkd-kaehm** [comment]
Hi,
@georgringer this stuff resolving the configuration from given Pid in the Page Tree. Also from deepest to superior Pages.
I guess it is needed by IndexService and other parts.

Also it is not for including TS.

**Comment by georgringer** [comment]
so whats the point of finding out the pages which hold a sys_template record?

**Comment by timohund** [comment]
@georgringer 

The reason is:

* In many parts the TypoScript is used in the backend context (To get the indexing configuratio, ...). This is allways fetched with the context page id, what makes sence because you can overwrite it everywhere in the three. Since this is a very expensive operation this it cached Util::initializeTsfe ...

But: The amount of cachehits is low since the configuration is fetched for every individual id. 
In the most systems this is not really required because you can just get the configuration from the closest page with a template on. Doing it like this results in much more cache hits. Since there are some conditions where this is not working (e.g. this conditions, it is possible to configure the behaviour: useConfigurationFromClosestTemplate in ext_conf_template.txt)

Just to get me right, i just want to explain the technical backrounds. If you have a idea to optimize or can create a patch we replace it with something simpler. 

So in short: Reason is to have more cacheHits since in most cases the closest template page can be used for the relevant typoscript template.

**Comment by georgringer** [comment]
not valid anymore? @dkd-kaehm
